README.md

```diff
@@ -22,15 +21,153 @@ various parts of Emmy library, please refer to additional documentation in the *
+ * The same certificate can be used to connect to different services (even if the databases are joined service providers 
+ cannot map/link the users)
+ * When connecting to a service, user can choose which data to reveal - some services might require only the possession 
+ of a certificate (e.g. certifying that user paid for something), others might require some subset of data contained in certificate
+ 
+While anonymity is obviously a MUST in e-voting, it might gradually become more important in other scenarios as well: 
+
+ * online subscriptions 
+ * wearable healthcare // diabetes - sending data to a diabetes research team 
+ * vehicle communications - cars sending out data about traffic
+ 
+### CL API
+ 
+Let's say University issues to each student a credential where the following attributes are written: name, gender, age,
+student status (graduate/postgraduate). University has its own public and secret key and plays the role of organization
+(when a student connects to the University, a new `Org` from `crypto/zkp/schemes/cl/org.go` is instantiated and 
+responsible for issuing a credential).
+
+First, a student needs to create a master secret key:
+
+```
+masterSecret := pubKey.GenerateUserMasterSecret() // pubKey is University public key
+```
+
+Now let's say variables `name`, `gender`, `age`, `studentStatus` contain big.Int representations for name, gender,
+age, studentStatus (support for different attribute is on TODO list). A student now creates `CredentialManager` which
+will interact with `Org` to obtain a credential (see test code in `client/cl_test.go`):
+
+```
+knownAttrs := []*big.Int{name, gender, age, studentStatus}
+committedAttrs := []*big.Int{}
+hiddenAttrs := []*big.Int{}
+credManager, err := cl.NewCredentialManager(params, pubKey, masterSecret, knownAttrs, committedAttrs,
+    hiddenAttrs)
+```
+
+Note that `committedAttrs` and `hiddentAttrs` are empty - these are needed for attributes for which the organization
+should know only commitments or nothing at all.
+
+Now a student instantiates a communication client:
+
+```
+clClient, err := NewCLClient(testGrpcClientConn)
+```
+
+Student now asks for credential:
+
+```
+cred, err := clClient.IssueCredential(credManager)
+```
+
+Now a student has a credential which can be used 
+
+
+
+ 
+### How does CL work
+ 
+ 
+
+# TODO
+
+ * Database layer
+ * Refactor Camenisch-Lysyanskaya scheme (database records, challenge generation ... )
+ * Additional proofs for Camenisch-Lysyanskaya scheme (range proof for attributes ... )
+ * Revocation for Camenisch-Lysyanskaya scheme
+ * Attribute types in Camenisch-Lysyanskaya scheme (string, int, date, enum)
+ * Performance optimization
+ * Efficient attributes for anonymous credentials [15]
+ * Camenisch-Lysyanskaya scheme based on pairings [14]
+ * Fix Camenisch-Shoup verifiable encryption
+ * Refactor proofs to set all parameters in a constructor (see comment in Schnorr prover)
+ 
 # Installation
```

Comment:
Can you please add this before Installation section (and also update the TOC):

# Warning
_All components of Emmy cryptography library are a work in progress. At this point, the library can be used to build proof of concept implementations for research purposes and **should never be used in production**. Project's code organization and library APIs are **not stable** - they are expected to undergo major changes, and may be changed at any point._